### DISCUSSION TITLE
ChemChaste compilation error

### DISCUSSION THREAD

**Discussion by fcentler** [Chaste/Chaste#308] [Q&A] [answered]
Hi, I am just getting started with Chaste / ChemChaste and was able to get Chaste running in a Docker environment on a Silicon MacBook (2 out of 402 tests failed). I then was able to set up a ChemChaste as a project as suggested in the ChemChaste documentation and was able to prepare the makefiles with cmake.

However, when I try to compile "TestHello" (or “TestChemChaste") using “make TestHello", after some successful compilation steps the build stops with the attached error.

Any help would be greatly appreciated!

Thanks & Cheers
Florian
![Image](https://github.com/user-attachments/assets/b9afea71-9a02-4d21-876a-089c8fa560db)

**Comment by fcooper8472** [comment, marked as answer]
Hi @fcentler,

The problem you're encountering is that the ChemChaste project is not compatible with the latest version of Chaste. The specific error you're running into relates to changes made here:
https://github.com/Chaste/Chaste/issues/151

@AlexFletcher might be able to advise on the most recent version of Chaste that ChemChaste is expected to work with, but it's possible you would run into other problems compiling an older version of Chaste on Apple Silicon.

Ultimately, the best solution would be to update ChemChaste to work with the latest version of Chaste. Here, for instance, you may just need to change that line to:

```cpp
MutableMesh<2,2>* p_mesh = p_generator->GetMesh().get();
```

but it's very possible there will be other changes necessary.

**Reply by fcentler** [reply]
Great, thanks a lot for looking into this!

Thanks & Cheers, Florian
